### ISSUE TITLE
Metacat will reject updated eml210 document if there is not rev 1 on it.

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#580]
---

Author Name: **Jing Tao** (Jing Tao)
Original Redmine Issue: 3701, https://projects.ecoinformatics.org/ecoinfo/issues/3701
Original Date: 2008-12-15
Original Assignee: Michael Daigle

---

Hi, Mike:

I have some issue in saving eml to metacat 1.9.0 release candidate. If i work a eml on morpho, see i created tao.12566.1 and saved it locally. Then I
modified it a little bit and tried to save them as tao.12566.2. Unfortunatley, I go the error in metacat:
 Error in writing eml document to the databaseDocument not found for Accession number tao.12566 [edu.ucsb.nceas.metacat.MetaCatServlet]
edu.ucsb.nceas.metacat.AccessionNumberException: Document not found for Accession number tao.12566.

I used the online eml parser to check the if tao.12566.2 is valid or not. The result shows it is a valid eml document.

I tried to use knb/style/skins/dev/devtool.html to update tao.12566.2 and got the same error. However, if i changed to insert action and gave it new id
tao.15.1, then the inserting worked even the eml doc is same.

This means if metacat doesn't have a docid, e.g, tao.12566.1, and you try to update a document tao.12566.2, it will fail. This is not convenient, since
user may work couple version locally, then save to metacat. Do you think we need to fix this issue?

I think metacat 1.8 doesn't have this behavior, but I not sure.

Thanks,

Jing

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Michael Daigle** (Michael Daigle)
Original Date: 2008-12-18T18:27:29Z

---

Can't reproduce this after release candidate 4.  Going to ask jing to try and reproduce it.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jing Tao** (Jing Tao)
Original Date: 2008-12-24T23:03:05Z

---

After disucssion with Matt and Mike, we decided morpho should use "INSERT" rather than "UPDATE" if there is no same id in metacat no matter the current version is 1 or greater than 1. Now morpho will check the docid existence status before inserting, then insert a document base on the status. The bug was fixed.
